### ISSUE TITLE
Worker services are not properly cleaned up

### ISSUE THREAD

**Issue by derekchiang** [pachyderm/pachyderm#1732]
In 1.4.5 we put a `Service` on top of each worker pool.  However, those services are not deleted when pipelines are deleted.

**Comment by jdoliner** [comment]
Do you remember what you did to reproduce this? I just tried this out and the `Service` seems to get deleted.

**Comment by derekchiang** [comment]
Actually, it turns out the problem is really that when you `make clean-launch-dev`, only the RCs but not the services are deleted.  The solution is to update the `clean-launch-dev` target to delete the services as well, which I will do in a PR.

**Comment by pappasilenus** [comment]
We are observing this again in 1.10.0.  Steps to reproduce:

1. Create a pipeline.  Note that a service is created called <pipeline-name>-v1
2. Update the pipeline. There are 2 separate services now, -v1 and -v2.  -v1 stays around, even after the -v1 version of the pod is terminated.

**Comment by pstrato-agrimetrics** [comment]
On pachyderm 1.9.11/azure, after a pipeline update I can see old k8s network services and endpoints. 
Here's an old service of a pipeline named transform:
````
apiVersion: v1
kind: Service
metadata:
  annotations:
    prometheus.io/port: "9090"
    prometheus.io/scrape: "true"
  creationTimestamp: "2020-04-03T11:08:52Z"
  labels:
    app: pipeline-transform-v1
    component: worker
    pipelineName: transform
    suite: pachyderm
  name: pipeline-transform-v1
  namespace: pachyderm
  resourceVersion: "20602943"
  selfLink: /api/v1/namespaces/pachyderm/services/pipeline-transform-v1
  uid: 8047371e-759b-11ea-9730-8aa0b82b0ac8
spec:
  clusterIP: 10.0.32.125
  ports:
  - name: grpc-port
    port: 80
    protocol: TCP
    targetPort: 80
  - name: prometheus-metrics
    port: 9090
    protocol: TCP
    targetPort: 9090
  selector:
    app: pipeline-transform-v1
    component: worker
    pipelineName: transform
    suite: pachyderm
  sessionAffinity: None
  type: ClusterIP
status:
  loadBalancer: {}
````
Here's old service of a pipeline named transform:
```
apiVersion: v1
kind: Endpoints
metadata:
  creationTimestamp: "2020-04-03T11:08:52Z"
  labels:
    app: pipeline-transform-v1
    component: worker
    pipelineName: transform
    suite: pachyderm
  name: pipeline-transform-v1
  namespace: pachyderm
  resourceVersion: "20606372"
  selfLink: /api/v1/namespaces/pachyderm/endpoints/pipeline-transform-v1
  uid: 804af131-759b-11ea-9730-8aa0b82b0ac8
```

the pods for V1 no longer exists after an update.

**Comment by msteffen** [comment]
I think this issue spans a long enough timeframe that it refers to different underlying problems, but we recently merged #4782, which is in Pachyderm >=1.10.3 and should fix this issue in all versions of pachyderm since then. If anyone affected sees stale k8s services after this version, please re-open this issue or file a new one.

I believe this is also a duplicate of https://github.com/pachyderm/pachyderm/issues/4071, which has also been closed.
